front/app/components/admin/PostManager/components/IdeaPreviewIndex.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import clHistory from 'utils/cl-router/history';
+import GoBackButton from 'components/UI/GoBackButton';
+import { Box } from '@citizenlab/cl2-component-library';
+import IdeaPostPreview from './PostPreview/Idea/IdeaPostPreview';
+
+interface Props {
+  goBackUrl: string;
+}
+
+type PreviewMode = 'view' | 'edit';
+
+const IdeaPreviewIndex = ({ goBackUrl }: Props) => {
+  const [previewMode, setPreviewMode] = useState<PreviewMode>('view');
+  const { ideaId } = useParams() as {
+    ideaId: string;
+  };
+
+  const handleOnClose = () => {
+    clHistory.push(goBackUrl);
+  };
+
+  const handleOnSwitchPreviewMode = () => {
+    setPreviewMode(previewMode === 'edit' ? 'view' : 'edit');
```

Comment:
Since the new state is going to be derived from the old one, it is better to use this option. By doing this, we ensure that the new state is derived from the previous state value instead of relying on the potentially stale previewMode variable. In this case it won't be stale but it helps to keep it robust from the beginning.
```suggestion
    setPreviewMode((prevPreviewMode) =>
      prevPreviewMode === 'edit' ? 'view' : 'edit'
    );
```